contracts/src/base/registry/facets/checker/EntitlementChecker.sol

```diff
@@ -153,6 +160,27 @@ contract EntitlementChecker is IEntitlementChecker, Facet {
     );
   }
 
+  /**
+   * @notice Get the nodes registered by an operator
+   * @param operator The address of the operator
+   * @return An array of node addresses
+   */
+  function getNodesByOperator(
+    address operator
+  ) external view returns (address[] memory) {
+    EntitlementCheckerStorage.Layout storage layout = EntitlementCheckerStorage
+      .layout();
+    uint256 nodeCount = layout.nodes.length();
+    address[] memory nodes = new address[](nodeCount);
+    for (uint256 i = 0; i < nodeCount; i++) {
+      nodes[i] = layout.nodes.at(i);
+      if (layout.operatorByNode[nodes[i]] == operator) {
+        nodes[i] = layout.nodes.at(i);
```

Comment:
This is the same as line 176? Were you trying to build a packed array of nodes here? I think that's one loop to count and one loop to assign.

Reply:
https://github.com/river-build/river/pull/171